app/scripts/modules/core/src/cluster/ClusterPostSearchResultSearcher.ts

```diff
@@ -14,38 +14,26 @@ import { IPostSearchResultSearcher } from 'core/search/searchResult/PostSearchRe
 
 export class ClusterPostSearchResultSearcher implements IPostSearchResultSearcher<IServerGroupSearchResult> {
 
-  private static TYPE = 'clusters';
-
   constructor(private $q: IQService, private $state: StateService) {}
 
   public getPostSearchResults(inputs: IServerGroupSearchResult[] = []): IPromise<ISearchResultSet[]> {
+    const type = 'clusters';
+
+    // create clusters based on the server group search results
+    const serverGroups = inputs.map((serverGroup: IServerGroupSearchResult) => {
+      const { account, application, cluster, provider, stack } = serverGroup;
+      const urlBuilder = urlBuilderRegistry.getBuilder(type);
+      const href = urlBuilder.build(Object.assign({ type }, serverGroup), this.$state);
+
+      return { account, application, cluster, displayName: cluster, href, provider, stack, type };
+    });
+
+    const clusters: IClusterSearchResult[] = uniqBy(serverGroups, 'cluster');
```

Comment:
This reads as if it would only return the first result of a cluster in a given account, omitting any duplicates. Worth verifying that's not actually happening, e.g. if you had "foo-bar-v001" in test and "foo-bar-v005" in prod, are you only getting one "foo-bar" cluster, or both?

Reply:
I think this is by design. The clusters results are derived from the servergroups

Reply:
switched from ` uniqBy(serverGroups, 'cluster');` to `sg => `uniqBy(serverGroups,${sg.account}-${sg.cluster})`